torch/testing/_internal/common_utils.py

```diff
@@ -1049,6 +1049,46 @@ def setUp(self):
 
         set_rng_seed(SEED)
 
+    def genSparseCSRTensor(self, size, nnz, device, dtype, index_dtype):
+        sparse_dim = 2
+        assert all(size[d] > 0 for d in range(sparse_dim)) or nnz == 0, 'invalid arguments'
+
+        def random_sparse(m, n, nnz_per_row):
+            rows = torch.arange(m, dtype=index_dtype).repeat(nnz_per_row)
+            cols = torch.randint(0, n, size=[nnz_per_row * m], dtype=index_dtype)
+            return rows, cols
+
+        def coo_to_csr(indices, dim):
+            r = torch.zeros(dim + 1, dtype=index_dtype)
+            last_i = 0
+            for i in indices:
+                if i == last_i:
+                    r[last_i + 1] += 1
+                else:
+                    for _i in range(last_i, i + 1):
+                        r[_i + 1] = r[last_i + 1]
+                    last_i = i
+                    r[last_i + 1] += 1
+            for _i in range(last_i, dim):
+                r[_i + 1] = r[last_i + 1]
+            return r
+
+        values = make_tensor([nnz], device=device, dtype=dtype, low=-1, high=1)
```

Comment:
Why such a tight range of values?

Reply:
Some tests from `test_sparse` expect tight ranges because some test  with default eps/tolerance will fail.  So I've just replicated that for genSparseCSRTensor.